template-extension.js

```diff
@@ -235,15 +235,29 @@ Template.prototype.copyAs = function (newTemplateName) {
 // that the field exists somewhere.
 Blaze.TemplateInstance.prototype.get = function (fieldName) {
   var template = this;
-
+  var result;
   while (template) {
-    if (fieldName in template) {
-      return template[fieldName];
+    result = hasProperty(template,fieldName);
+    if (result) {
+      return result;
     }
     template = template.parent(1, true);
   }
 };
 
+
+function hasProperty(obj, prop) {
+  var parts = prop.split(".");
+  var cur = obj;
+  for (var i=0; i<parts.length; i++) {
+    if (!cur[parts[i]])
```

Comment:
Agree, should be something like `if (!cur.hasOwnProperty(parts[i]))`

And add an additional test where the value is `false` to make sure it returns `false`.